src/utils/getResponse.ts

```diff
@@ -33,27 +49,7 @@ export const getResponse = async <
     }
   }
 
-  const result = await relevantHandlers.reduce<
-    Promise<RequestHandlerExecutionResult<any> | null>
-  >(async (acc, handler) => {
-    const previousResults = await acc
-
-    if (!!previousResults?.response) {
-      return acc
-    }
-
-    const result = await handler.run(request)
-
-    if (result === null || !result.response || result.handler.shouldSkip) {
-      return null
-    }
-
-    if (result.response.once) {
-      handler.markAsSkipped(true)
-    }
-
-    return result
-  }, Promise.resolve(null))
+  const result = await getResult(relevantHandlers, request)
```

Comment:
This is a great intention, but I would leave the previous logic as-is. This part has some history behind it, changing it may introduce regressions not related to the issue we are solving. 

Let's leave everything as-is, add a new failing test, and then address the issue. What do you think?